lib/internal/per_context/domexception.js

```diff
@@ -46,13 +50,23 @@ const disusedNamesSet = new SafeSet()
   .add('NoDataAllowedError')
   .add('ValidationError');
 
-class DOMException {
+class DOMException extends JSTransferable {
   constructor(message = '', name = 'Error') {
-    ErrorCaptureStackTrace(this);
-    internalsMap.set(this, {
-      message: `${message}`,
-      name: `${name}`
-    });
+    super();
+
+    let code;
+    if (disusedNamesSet.has(name)) {
+      code = 0;
+    } else {
+      code = nameToCodeMap.get(name);
+      code = code === undefined ? 0 : code;
+    }
+
+    const o = { __proto__: null };
+    ErrorCaptureStackTrace(o);
+    FunctionPrototypeCall(this[kDeserialize],
+                          this,
+                          { name, message, code, stack: o.stack });
```

Comment:
Wouldn't this be equivalent and simpler?
```suggestion
    this[kDeserialize]({ name, message, code, stack: o.stack });
```